lib/ansible/modules/system/osx_defaults.py

```diff
@@ -15,413 +16,975 @@
+            return True
+    return False
 
-        # Ensure the value is the correct type
-        self.value = self._convert_type(self.type, self.value)
 
-    # /init --------------------------------------------------------------- }}}
+class Data(binary_type):
+    """
+    Object representing binary data.
+    Instance should be initialize with binary data encoded in base64 codec.
+    """
+    # List of all Base64 accepted characters.
+    BASE64_CHARS = string.ascii_letters + string.digits + '+/='
 
-    # tools --------------------------------------------------------------- {{{
-    """ Converts value to given type """
+    # List of all text characters.
+    TEXT_CHARS = bytearray(set([7, 8, 9, 10, 12, 13, 27]) | set(range(0x20, 0x100)) - set([0x7f]))
 
-    def _convert_type(self, type, value):
+    def __new__(cls, data):
+        if isinstance(data, cls):
+            return data
 
-        if type == "string":
-            return str(value)
-        elif type in ["bool", "boolean"]:
-            if isinstance(value, (binary_type, text_type)):
-                value = value.lower()
-            if value in [True, 1, "true", "1", "yes"]:
-                return True
-            elif value in [False, 0, "false", "0", "no"]:
-                return False
-            raise OSXDefaultsException("Invalid boolean value: {0}".format(repr(value)))
-        elif type == "date":
+        # Try to convert unicode to ascii string.
+        if isinstance(data, text_type):
             try:
-                return datetime.datetime.strptime(value.split("+")[0].strip(), "%Y-%m-%d %H:%M:%S")
-            except ValueError:
-                raise OSXDefaultsException(
-                    "Invalid date value: {0}. Required format yyy-mm-dd hh:mm:ss.".format(repr(value))
-                )
-        elif type in ["int", "integer"]:
-            if not str(value).isdigit():
-                raise OSXDefaultsException("Invalid integer value: {0}".format(repr(value)))
-            return int(value)
-        elif type == "float":
-            try:
-                value = float(value)
-            except ValueError:
-                raise OSXDefaultsException("Invalid float value: {0}".format(repr(value)))
-            return value
-        elif type == "array":
-            if not isinstance(value, list):
-                raise OSXDefaultsException("Invalid value. Expected value to be an array")
-            return value
+                data = data.encode('ascii')
+            except UnicodeEncodeError:
+                pass
 
-        raise OSXDefaultsException('Type is not supported: {0}'.format(type))
+        # Check if data is a valid base64 string. Short strings are not
+        # considered as binary.
+        if isinstance(data, binary_type) and len(data) > 51 and not data.translate(None, cls.BASE64_CHARS):
+            try:
+                binary_data = b64decode(data)
```

Comment:
Because some really normal character strings are valid base64-encoded binary data. Ex.:

`>>> b64decode('The quick brown fox jumps over the lazy dogs')
b"N\x17\xaa\xba'$n\xba0\x9d\xfa1\x8e\xe9\xa9\xb2\x8b\xde\xae\xd8^\x95\xac\xf2v\x88,"
`